neuro_san/internals/run_context/langchain/core/langchain_run_context.py

```diff
@@ -653,11 +663,16 @@ async def delete_resources(self, parent_run_context: RunContext = None):
         :param parent_run_context: A parent RunContext perhaps the same instance,
                         but perhaps not.  Default is None
         """
+
+        # Release model related resources:
+        if self.llm_resources:
+            await self.llm_resources.release_resources()
+
```

Comment:
This is where we actually release model-related resources when we clean up our RunContext.